etna/models/nn/chronos/base.py

```diff
@@ -0,0 +1,273 @@
+import os
+import warnings
+import zipfile
+from abc import abstractmethod
+from pathlib import Path
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Sequence
+from typing import Union
+from urllib import request
+
+import pandas as pd
+
+from etna import SETTINGS
+from etna.datasets import TSDataset
+from etna.distributions import BaseDistribution
+from etna.libs.chronos import BaseChronosPipeline
+from etna.libs.chronos import ChronosBoltModelForForecasting
+from etna.libs.chronos import ChronosModelForForecasting
+from etna.models.base import PredictionIntervalContextRequiredAbstractModel
+
+if SETTINGS.chronos_required:
+    import torch
+
+
+class ChronosBaseModel(PredictionIntervalContextRequiredAbstractModel):
+    """Base class for Chronos-like pretrained models."""
+
+    def __init__(
+        self,
+        model_name: str,
+        encoder_length: int,
+        device: str,
+        dtype: torch.dtype,
+        cache_dir: Path,
+        from_s3: bool,
+    ):
+        """
+        Init Chronos-like model.
+
+        Parameters
+        ----------
+        model_name:
+            Model name. See ``pretrained_model_name_or_path`` parameter of :py:func:`transformers.PreTrainedModel.from_pretrained`.
+        encoder_length:
+            Number of last timestamps to use as a context.
+        device:
+            Device type. See ``device_map`` parameter of :py:func:`transformers.PreTrainedModel.from_pretrained`.
+        dtype:
+            Torch dtype of computation. See ``torch_dtype`` parameter of :py:func:`transformers.PreTrainedModel.from_pretrained`.
+        cache_dir:
+            Local path to save model from huggingface during first model initialization. All following class initializations appropriate model version will be downloaded from this path.
+            See ``cache_dir`` parameter of :py:func:`transformers.PreTrainedModel.from_pretrained`.
+        from_s3:
+            Whether to load from s3 or huggingface. Mostly for developer usage.
+        """
+        super().__init__()
+        self.model_name = model_name
+        self.encoder_length = encoder_length
+        self.device = device
+        self.dtype = dtype
+        self.cache_dir = cache_dir
+        self.from_s3 = from_s3
+
+        if self.from_s3:
+            self._download_model_from_s3()
+            model_path = Path(self.cache_dir) / self.model_name
+        else:
+            model_path = Path("amazon") / self.model_name
+
+        self.pipeline = BaseChronosPipeline.from_pretrained(
```

Comment:
Maybe we can add **kwargs parameter passed to `from_pretrained` as in original chronos api. But I don't really know what parameters can be useful.